Assembler.UnitTests/EndMessageHandlerTests.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Assembler.Base;
+using Assembler.Core;
+using Assembler.Core.Entities;
+using Assembler.Core.Enums;
+using Moq;
+using NUnit.Framework;
+
+namespace Assembler.UnitTests
+{
+    [TestFixture]
+    public class EndMessageHandlerTests
+    {
+        private Mock<ITimeBasedCache<BaseMessageInAssembly>> _cacheMock;
+        private Mock<IFactory<BaseFrame, string>> _identifierFactoryMock;
+        private Mock<IMessageEnricher<BaseFrame, BaseMessageInAssembly>> _enricherMock;
+        private Mock<ICreator<BaseMessageInAssembly>> _messageInAssemblyCreatorMock;
+
+        private List<BaseMessageInAssembly> _assembledMessages;
+        private string _identifierString;
+
+        [SetUp]
+        public void Setup()
+        {
+            _cacheMock = new Mock<ITimeBasedCache<BaseMessageInAssembly>>();
+            _enricherMock = new Mock<IMessageEnricher<BaseFrame, BaseMessageInAssembly>>();
+            _messageInAssemblyCreatorMock = new Mock<ICreator<BaseMessageInAssembly>>();
+            _identifierString = Utilities.GetIdentifierString();
+            _identifierFactoryMock = Utilities.GetIdentifierMock();
+
+            _assembledMessages = new List<BaseMessageInAssembly>();
+        }
+
+        [TearDown]
+        public void Teardown()
+        {
+            _identifierFactoryMock.VerifyNoOtherCalls();
+            _enricherMock.VerifyNoOtherCalls();
+            _cacheMock.VerifyNoOtherCalls();
+            _messageInAssemblyCreatorMock.VerifyNoOtherCalls();
+        }
+
+        [TestCase(true)]
+        [TestCase(false)]
+        public void Handle_IdentifierInCache_MessageBeingRemovedEnrichedAndRaised(bool isToReleaseSingleEndFrame)
+        {
+            // Arrange
+            var frame = new Mock<BaseFrame>(FrameType.Initial);
```

Comment:
Not exactly critical, but perhaps you would like to change `FrameType.Initial` to `FrameType.End`, considering that would be the actual value

Reply:
Also appears in lines 87, 120, 143

Reply:
f

Reply:
Well, technically not, but see https://github.com/reverser-gever/Assembler/pull/2#discussion_r397141849

Reply:
👍